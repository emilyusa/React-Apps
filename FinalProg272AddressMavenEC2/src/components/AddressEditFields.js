import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import withStyles from '@material-ui/core/styles/withStyles';

const styles = theme => ({
    button: {
        margin: theme.spacing.unit
    },
    container: {
        flexGrow: 1,
        textAlign: 'center'
    },
    rootBar: theme.mixins.gutters({
        paddingTop: 16,
        paddingBottom: 16,
        paddingLeft: 8,
        paddingRight: 8,
        marginTop: theme.spacing.unit * 3,
        marginLeft: theme.spacing.unit * 3,
        marginRight: theme.spacing.unit * 3
    })
});

class AddressEditFields extends Component {
    render() {
        return (
            <div>
                <TextField
                    autoFocus
                    margin="dense"
                    id="name"
                    label="First Name"
                    type="string"
                    value={this.props.address.firstName}
                    onChange={e =>
                        this.props.addressChangedByUser('firstName', e)
                    }
                    fullWidth
                />
                <TextField
                    autoFocus
                    margin="dense"
                    id="lastname"
                    label="Last Name"
                    type="string"
                    value={this.props.address.lastName}
                    onChange={e =>
                        this.props.addressChangedByUser('lastName', e)
                    }
                    fullWidth
                />
            </div>
        );
    }
}

AddressEditFields.propTypes = {
    addressChangedByUser: PropTypes.func,
    address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string
    })
};

export default withStyles(styles)(AddressEditFields);
