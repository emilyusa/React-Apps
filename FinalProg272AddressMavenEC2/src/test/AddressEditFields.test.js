import React from 'react';
import { configure, shallow } from 'enzyme/build/index';
import Adapter from 'enzyme-adapter-react-16/build/index';
import AddressEditFields from '../components/AddressEditFields';

configure({ adapter: new Adapter() });

describe('AddressEditFields test', function() {
    const addresses = [
        {
            firstName: 'unknown',
            lastName: 'unknown'
        },
        {
            firstName: 'Chunyan',
            lastName: 'Li'
        }
    ];

    it('renders and displays the default first name', () => {
        const wrapper = shallow(
            <AddressEditFields addressList={addresses[0]} />
        );
        expect(wrapper.prop('addressList').firstName).toBe('unknown');
    });

    it('renders and displays the default last name', () => {
        const wrapper = shallow(
            <AddressEditFields addressList={addresses[0]} />
        );
        expect(wrapper.prop('addressList').lastName).toBe('unknown');
    });

    it('renders and displays the default first name after edit', () => {
        const wrapper = shallow(
            <AddressEditFields addressChangedByUser={beforeEach} />
        );
        expect(wrapper.prop('addressChangedByUser').length).toBe(0);
    });
});
