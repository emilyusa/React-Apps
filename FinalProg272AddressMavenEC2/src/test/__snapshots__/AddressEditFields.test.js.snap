// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressEditFields test should take an AddressEditFields snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AddressEditFields)
    addressList={
        Object {
            "firstName": "unknown",
            "lastName": "unknown",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addressList": Object {
        "firstName": "unknown",
        "lastName": "unknown",
      },
      "classes": Object {
        "button": "AddressEditFields-button-1",
        "container": "AddressEditFields-container-2",
        "rootBar": "AddressEditFields-rootBar-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addressList": Object {
          "firstName": "unknown",
          "lastName": "unknown",
        },
        "classes": Object {
          "button": "AddressEditFields-button-1",
          "container": "AddressEditFields-container-2",
          "rootBar": "AddressEditFields-rootBar-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
